### DISCUSSION TITLE
프로덕트 스펙 정하기 - 1차

### DISCUSSION THREAD

**Discussion by jayjeong8** [EasyAndBeauty/DailyReceipt-Flutter#4] [Ideas]
### 배경
적절한 범위로 프로젝트를 계획해서 열정을 잃지 않고 효과적으로 진행하고 싶습니다. 
모두가 출근이나 취준이라는 상황 속에 **부담없는 규모**로 진행할 수 있도록 1차 스펙을 제안합니다.

### 제안
아래 UI 이미지 예시의 핵심 기능만 구현하는 것을 1차 목표로 삼으면 좋을 것 같습니다.
- TODO 페이지 - TODO CRUD / 날짜 선택(일주일 단위, 월 단위)
- 영수증 출력 페이지 - 복사 / 저장
<img width="649" alt="image" src="https://github.com/EasyAndBeauty/DailyReceipt-Flutter/assets/82160479/3d83771b-7b39-440c-b7de-f113f619cebf">

기존 스펙은 여기에 더해 타이머, 마이 영수증(pin),  로그인 등이 있었으나 우선 위와 같은 스펙으로 진행한 뒤 우선순위에 따라서 추가해나가면 좋겠습니다!
- [피그마에서 이전 스펙 보기](https://www.figma.com/file/FbrfbsxJkyQkN1p9N8DYMU/daily-receipt?type=design&node-id=0%3A1&mode=design&t=9LvjVqCeNFxNeuYG-1)

이와 같이 진행할 경우 영수증에 TODO 항목 가격으로 표시할 **'타이머 누적시간'이 없으므로 'TODO 완료' 시간으로 표시**하는 것도 좋을 것 같습니다! 

많은 의견 부탁드립니다😊

**Comment by IGhost-P** [comment]
개인적으로 TODO page만 만드는걸 1차로 잡았으면 합니다~, 작업자가 여러명이긴해도, 다같이 플러터 CRUD를 경험해보았으면 해욥!

**Reply by jayjeong8** [reply]
오 동의합니다!! 작은 주기로 회고하면서 진행하면 동기 부여하기에도 더 좋을 것 같아요!

**Reply by IGhost-P** [reply]
작은 주기로 리팩터링을 같이 고민하는것도 좋을것 같아요!
최대한 좋은 코드를 유지하면서 개발하되, 계획했던 일정까지는 꼭 개발을 하고, 그러면서 생긴 부채는 n차 목표가 끝날때마다 2~3일 정도 기술부채나 리팩터링을 고민하면 좋겠어요!

**Reply by gygy7151** [reply]
이부분 넘 좋은 의견이라구 느낍니다!!
우선은 CRUD만 구현하구 리팩터링 같이 고민하면
재밌을꺼 같아요!

여담이지만 최근에 Todo mate라는 앱을 추천받아 사용하게 되었는데, 팔로우 기능통해서 친구들끼리 투두리스트 달성완료하면 알람으로 알려줘서 동기부여 제대로 되더라구요! 저희도 비슷하게 나중에 한번 만들어보면 좋을것 같다구 느꼈어요!

**Reply by jayjeong8** [reply]
두분 의견 감사합니다!! 

그러면 TODO 페이지 - TODO CRUD / 날짜 선택(일주일 단위, 월 단위) 기준으로 이슈 생성해둘게요! 
TODO 데이터 설계할 때 날짜 같이 가져가면 좋을 것 같아서 댄이 얘기한 TODO 페이지 기준으로 1차 기준 잡겠습니다✨

그리고 블루가 얘기해줘서 생각난건데 나중에 다른 사람의 영수증을 볼 수 있는 기능이 생겨도 좋을 것 같네요~
공유하기 선택한 항목만 공개하도록 하면 혼자 볼 항목이랑 공개할 항목이랑 나눠볼 수도 있을 것 같아요!

**Reply by gygy7151** [reply]
@jayjeong8 그쵸 영수증 팔로잉한 사람들끼리 서로 볼 수 있는 기능이 생겨도 좋을꺼 같아요!!!
영수증 생성되면 팔로워들한테 알람도 가구!!

**Reply by mdgarden** [reply]
> 그리고 블루가 얘기해줘서 생각난건데 나중에 다른 사람의 영수증을 볼 수 있는 기능이 생겨도 좋을 것 같네요~

두 분 요 아이디어 넘 좋네요! 투두메이트 저도 예전에 써봤었는데 재밌었어요 기회 되면 만들어보고 싶네욥
작은 규모로 주기적인 회고를 진행하는 것도 너무 좋은 것 같습니다🙆🙆

**Comment by mdgarden** [comment]
메모)
현단계에서 구현하지 않는 기능
- 로그인
- pin
- 타이머

**Reply by IGhost-P** [reply]
로그인이랑, 서버 개발을 염두하지 않는다는것은 사용자의 todo 정보를 로컬 DB에 저장한다는 의미일까요?

**Reply by mdgarden** [reply]
네! 저도 지금 로컬 DB/메모리에 대해서 정확하게 파악을 하지 않은 상태에서 드리는 말씀이긴한데, 디바이스 자체 메모리에 저장하는 방식으로 진행하고 싶습니다.
만약에 앱 출시 이후 반응이 좋아서 추가적으로 개발을 하게 되면(행복회로....ㅎㅎㅎ ㅠㅠ ) 그 때 별도 서버 대응을 해도 괜찮을 것 같아요
